lib/msal-browser/src/utils/BrowserUtils.ts

```diff
@@ -3,154 +3,162 @@
+        throw createBrowserAuthError(BrowserAuthErrorCodes.redirectInIframe);
     }
+}
 
-    /**
-     * Block redirectUri loaded in popup from calling AcquireToken APIs
-     */
-    static blockAcquireTokenInPopups(): void {
-        // Popups opened by msal popup APIs are given a name that starts with "msal."
-        if (BrowserUtils.isInPopup()) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.blockNestedPopups
-            );
-        }
+/**
+ * Block redirectUri loaded in popup from calling AcquireToken APIs
+ */
+export function blockAcquireTokenInPopups(): void {
+    // Popups opened by msal popup APIs are given a name that starts with "msal."
+    if (isInPopup()) {
+        throw createBrowserAuthError(BrowserAuthErrorCodes.blockNestedPopups);
     }
+}
 
-    /**
-     * Throws error if token requests are made in non-browser environment
-     * @param isBrowserEnvironment Flag indicating if environment is a browser.
-     */
-    static blockNonBrowserEnvironment(isBrowserEnvironment: boolean): void {
-        if (!isBrowserEnvironment) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.nonBrowserEnvironment
-            );
-        }
+/**
+ * Throws error if token requests are made in non-browser environment
+ * @param isBrowserEnvironment Flag indicating if environment is a browser.
+ */
+export function blockNonBrowserEnvironment(
+    isBrowserEnvironment: boolean
+): void {
+    if (!isBrowserEnvironment) {
+        throw createBrowserAuthError(
+            BrowserAuthErrorCodes.nonBrowserEnvironment
+        );
     }
+}
 
-    /**
-     * Throws error if initialize hasn't been called
-     * @param initialized
-     */
-    static blockAPICallsBeforeInitialize(initialized: boolean): void {
-        if (!initialized) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.uninitializedPublicClientApplication
-            );
-        }
+/**
+ * Throws error if initialize hasn't been called
+ * @param initialized
+ */
+export function blockAPICallsBeforeInitialize(initialized: boolean): void {
+    if (!initialized) {
+        throw createBrowserAuthError(
+            BrowserAuthErrorCodes.uninitializedPublicClientApplication
+        );
     }
 }
+
+/**
+ * Adds a preconnect link element to the header which begins DNS resolution and SSL connection in anticipation of the /token request
+ * @param loginDomain Authority domain, including https protocol e.g. https://login.microsoftonline.com
+ * @returns
+ */
+export function preconnect(authority: string): void {
+    const link = document.createElement("link");
+    link.rel = "preconnect";
+    link.href = new URL(authority).origin;
+    link.crossOrigin = "anonymous";
+    document.head.appendChild(link);
+
+    // The browser will close connection if not used within a few seconds, remove element from the header after 10s
```

Comment:
Does the 10s timeout align with the browser implementation?  I'm not sure if the browser were to close after 3 seconds, would this implementation open then close several times before the preconnect is removed?

Reply:
If I'm parsing this question correctly - The connection to the STS is opened when this preconnect element is added to the page which we only do once per request. If the browser chooses to close the connection after 3 seconds it will stay closed until another preconnect element is added (another ssoSilent request is made) so this 10s timeout is just cleaning up dead elements.

Also of note - if a connection already exists to a given domain when a preconnect element is added it is a no-op.